server/src/main/java/org/elasticsearch/ElasticsearchException.java

```diff
@@ -407,7 +423,29 @@ public static ElasticsearchException fromXContent(XContentParser parser) throws
         return innerFromXContent(parser, false);
     }
 
+    /**
+     * Generate a {@link ElasticsearchException} from a {@link XContentParser}. This does not
+     * return the original exception type (ie NodeClosedException for example) but just wraps
+     * the type, the reason and the cause of the exception. It also recursively parses the
+     * tree structure of the cause, returning it as a tree structure of {@link ElasticsearchException}
+     * instances.
+     *
+     * @param parser the parser
+     * @param retainXContentFields if true, then the exception reason and type will be retained for
+     *                             re-serialization to XContent.
+     */
+    public static ElasticsearchException fromXContent(XContentParser parser, boolean retainXContentFields) throws IOException {
```

Comment:
I think that ideally we should end up not specifying whether to retain or not, it should be default. But we might need this in 7.x if the message needs to stay BWC?